bin/alignment.py

```diff
@@ -196,16 +196,17 @@ def processData(json_obj):
         samples_data)
     output_json = json.dumps(output_dict)
     return output_json
+# read input till the EOF
+for line in iter(sys.stdin.readline,''):
+    input_json += line
 
+# process data
+json_obj = json.loads(input_json)
+processedData = processData(json_obj)
 
-for line in iter(sys.stdin.readline, ''):
-    if 'start processing' in line:
-        continue
-    if 'end processing' in line:
-        json_obj = json.loads(input_json)
-        processedData = processData(json_obj)
-        print processedData
-        sys.stdout.flush()
-        print 'stop'
-        sys.stdout.flush()
-    input_json += line
+# sent back processed data
+sys.stdout.write(processedData)
+sys.stdout.flush()
+
+# must exit the program manully
+sys.exit(0)
```

Comment:
code should  be made to exit from other places too. e.g. if the algorithm fails, then what happens ? Should the code not exit in that case ?

Reply:
@rish9511 No matter where python will fail, all thing will be handled in Qt.